dspace-api/src/main/java/org/dspace/identifier/ClarinVersionedDOIIdentifierProvider.java

```diff
@@ -0,0 +1,271 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.identifier;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.DSpaceObject;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataValue;
+import org.dspace.core.Context;
+import org.dspace.identifier.doi.DOIConnector;
+import org.dspace.identifier.doi.DOIIdentifierException;
+import org.dspace.services.ConfigurationService;
+import org.dspace.versioning.VersionHistory;
+import org.dspace.versioning.service.VersionHistoryService;
+import org.dspace.versioning.service.VersioningService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This class is copied from the VersionedDOIIdentifierProvider. The main difference is that was removed code
+ * where is created the handle based on the history.
+ *
+ * @author Milan Majchrak (milan.majchrak at dataquest.sk)
+ * @author Marsa Haoua
+ * @author Pascal-Nicolas Becker (dspace at pascal dash becker dot de)
+ */
+public class ClarinVersionedDOIIdentifierProvider extends DOIIdentifierProvider {
+    /**
+     * log4j category
+     */
+    private static final Logger log = LogManager.getLogger(VersionedDOIIdentifierProvider.class);
+
+    protected DOIConnector connector;
+
+    static final char DOT = '.';
+    protected static final String pattern = "\\d+\\" + String.valueOf(DOT) + "\\d+";
+
+    @Autowired(required = true)
+    protected VersioningService versioningService;
+    @Autowired(required = true)
+    protected VersionHistoryService versionHistoryService;
+
+    @Override
+    public String mint(Context context, DSpaceObject dso)
+            throws IdentifierException {
```

Comment:
quite complicated. no chance to make it more readable?

Reply:
This class is copied from the VersionedDOIIdentifierProvider. I've updated just a few rows. This mint method is complicated for me too but I wouldn't like to change it if it is not necessary.
I copied it to another class for better maintenance in the future .